app/controllers/items_controller.rb

```diff
@@ -1,5 +1,7 @@
 class ItemsController < ApplicationController
   before_action :authenticate_user!, except: [:show, :index]
+  before_action :set_item, only: [:show, :edit, :update]
+  before_action :set_redirect_to_root, only: [:edit, :create]
```

Comment:
```suggestion
  before_action :set_redirect_to_root, only: [:edit, :update]
```
createアクションは出品時に使用するものですので、このメソッドは適用されません。
updateアクションに変更しましょう。